test/strings.js

```diff
@@ -59,6 +59,22 @@ describe('strings', () => {
       expect(inspect('foobarbaz', { truncate: 3 })).to.equal("'…'")
     })
 
+    it('truncates strings involving surrogate pairs longer than truncate (7)', () => {
+      expect(inspect('🐱🐱🐱', { truncate: 7 })).to.equal("'🐱🐱…'") // not '🐱🐱\ud83d…'
```

Comment:
I'm the author of fast-check: a property based testing library. This case is well-suited for such tests, so I have to at least let you know of it. Test would be something like:

```js
it('truncates strings into valid unicode strings', () => {
  fc.assert(fc.property(fc.fullUnicodeString(), fc.nat(), (text, truncate) => {
    expect(() => encodeURI(inspect(text, { truncate }))).not.to.throw()
  }))
})
```

And it would cover this specific case but possibly other that might came one day.

Reply:
Sounds awsome! Is there a chai integration 😉?

Reply:
No dedicated chai integration. At least nothing deeper than the suggestion I made 😅

I had a chat with maintainers of Jest in the past to wrap `assert`+`property` into some kind of `expect(...).toSomething(...)` but I/we never pushed the idea further. The only integration I have are mostly connecting it/test of the test runners with the library. But nothing at the level of expect itself.

Reply:
@keithamus let me know if the technique might be interesting for loupe/chaijs? If you feel it might I can try to propose a pull request with some tests using the technique (I can work on such PR even if it never gets merged)